### PR TITLE
Refactor features

### PR DESCRIPTION
**Pull request by waldyr**
I've refactored the features to use the spec/samples files. Now samples folder might need a better place to stay.

Thinking on doing: `Given(/a configuration file called '(.+)'$/)` and search for the specific file in samples/configurations removing the need from `strip_heredoc`.

@troessner @mvz Wdyt?

### CODE REVIEW

File: features/programmatic_access.feature [hunk 1]
```diff
@@ -1,4 +1,5 @@
 Feature: Using Reek programmatically
+        [5]:UncommunicativeVariableName: Smelly#x has the variable name 'y' [https://github.com/troessner/reek/blob/master/docs/Uncommunicative-Variable-Name.md]
```
Comment (troessner) [inline features/programmatic_access.feature]: Pretty sure this doesnt belong here ;)

File: features/command_line_interface/basic_usage.feature [hunk 2]
```diff
@@ -9,13 +9,7 @@ Feature: The Reek CLI maintains backwards compatibility
...
-        [1]:IrresponsibleModule: Dirty has no descriptive comment [https://github.com/troessner/reek/blob/master/docs/Irresponsible-Module.md]
-        [3]:LongParameterList: Dirty#awful has 4 parameters [https://github.com/troessner/reek/blob/master/docs/Long-Parameter-List.md]
-        [3]:UncommunicativeParameterName: Dirty#awful has the parameter name 'x' [https://github.com/troessner/reek/blob/master/docs/Uncommunicative-Parameter-Name.md]
-        [5]:UncommunicativeVariableName: Dirty#awful has the variable name 'w' [https://github.com/troessner/reek/blob/master/docs/Uncommunicative-Variable-Name.md]
-        [3]:UnusedParameters: Dirty#awful has unused parameter 'log' [https://github.com/troessner/reek/blob/master/docs/Unused-Parameters.md]
-        [3]:UnusedParameters: Dirty#awful has unused parameter 'offset' [https://github.com/troessner/reek/blob/master/docs/Unused-Parameters.md]
-        [3]:UnusedParameters: Dirty#awful has unused parameter 'y' [https://github.com/troessner/reek/blob/master/docs/Unused-Parameters.md]
+      demo.rb -- 2 warnings:
```
Comment (troessner) [inline features/command_line_interface/basic_usage.feature]: Yes, love it already.

**Comment by troessner** [comment]
>> Thinking on doing: Given(/a configuration file called '(.+)'$/) and search for the specific file in samples/configurations removing the need from strip_heredoc.

👍  from my side. the only reason why we didn't do this before was because then we would have a lot of different config files, which would become unwieldy. With those changes above in place I'm all up for it. @mvz ?

>> I've refactored the features to use the spec/samples files. Now samples folder might need a better place to stay.

Agree.

File: features/configuration_files/masking_smells.feature [hunk 3]
```diff
@@ -22,28 +22,34 @@ Feature: Masking smells using config files
     Then it succeeds
     And it reports nothing
 
-  Scenario: allow masking some calls for duplication smell
+  Scenario: provide extra masking inline in comments
```
Comment (mvz) [inline features/configuration_files/masking_smells.feature]: Can you put these scenarios in their original order? I'm having a hard time seeing what changed here.

**Comment by mvz** [comment]
Yes, it makes sense to do the same with configuration files as with ruby code sample files.

File: features/configuration_files/masking_smells.feature (hunk 3 above)
Comment (waldyr) [inline features/configuration_files/masking_smells.feature]: Sure :)

**Comment by waldyr** [comment]
@troessner @mvz I Think I'm done here. Wdyt?

File: features/samples.feature [hunk 4]
```diff
@@ -4,9 +4,9 @@ Feature: Basic smell detection
   I want to detect the smells in my Ruby code
 
   Scenario: Correct smells from inline.rb
-    Given the "inline.rb" sample file exists
-    And the "optparse.rb" sample file exists
-    And the "redcloth.rb" sample file exists
+    Given the smelly file 'inline.rb'
+    Given the smelly file 'optparse.rb'
```
Comment (troessner) [inline features/samples.feature]: Should be "And" here and below.

**Comment by troessner** [comment]
One last tiny change and this one is good to go! You can squash now as well.

**Comment by waldyr** [comment]
Done.

**Comment by troessner** [comment]
Merged! Awesome job @waldyr !

**Pull request merged**